simple_node.go

```diff
@@ -88,6 +89,22 @@ func (node *SimpleNode) Equals(node2 Node) bool {
 		return false
 	}
 
+	useAncestrySourceMatching := flag.Lookup("ancestry-source-matching").Value.String() //indexes a map CommandLine.formal
```

Comment:
You cannot lookup flags like this. While this might work when using it through the CLI, this can also be used a library. It's also just a leaky abstraction to write it this way. You have three choices:

1. Make the extra comparison logic permanent. I'm fine with that - but there are still caveats to consider below.
2. If you want/need to control the logic you will need to provide that through another Equals method with options, for example:

```go
type SimpleNodeEqualsOptions struct {
    // UseAncestrySourceMatching ...
    UseAncestrySourceMatching bool
}

func (node *SimpleNode) Equals(node2 Node) bool {
    return node.EqualsOptions(node2, SimpleNodeEqualsOptions{})
}

func (node *SimpleNode) EqualsOptions(node2 Node, o SimpleNodeEqualsOptions) bool {
    // ...

    if o.UseAncestrySourceMatching && node.Tag().String() == "Source" && tag.String() == "Source" {
    }
}
```

3. This is probably the best option for you, use a `FilterFunction` (https://github.com/elliotchance/gedcom/blob/9061dcabbb6bdd35ed84163225bbb4737b2ba3ab/filter.go#L36) with a CLI argument.

`SimpleNode.Equals` only will only apply to nodes that do not have a custom type with an `Equals` method of their own. This is an important consideration because this is generic logic that you may want to apply to all nodes _on top_ of any logic.

If you want this to apply globally (and be controlled from the CLI) you might find it easier to do preprocessing first instead. That is, based on a CLI flag, you preprocess the files so when the comparison happens the logic has already been normalized.

We do in this already using FilterFunctions. For example the `-only-official` CLI flag (https://github.com/elliotchance/gedcom/blob/master/filter_flags.go#L57-L58) triggers the files to remove non-official GEDCOM tags so they don't exist for the comparison: https://github.com/elliotchance/gedcom/blob/9061dcabbb6bdd35ed84163225bbb4737b2ba3ab/filter.go#L143. That file also contains some other filters examples.